packages/configure-splash-screen/src/cli-command.ts

```diff
@@ -0,0 +1,196 @@
+import { Command } from 'commander';
+import path from 'path';
+
+import configureAndroid from './android';
+import configureIos from './ios';
+import {
+  SplashScreenImageResizeMode,
+  SplashScreenStatusBarStyle,
+  Platform,
+  PlatformType,
+} from './constants';
+import { AndroidSplashScreenJsonConfig, IosSplashScreenJsonConfig } from './types';
+import { validateEnumValue } from './validators';
+
+const AVAILABLE_OPTIONS_NAMES = [
+  'imageResizeMode',
+  'platform',
+  'backgroundColor',
+  'imagePath',
+  'darkModeBackgroundColor',
+  'darkModeImagePath',
+  'statusBarHidden',
+  'statusBarStyle',
+  'darkModeStatusBarStyle',
+  'statusBarTranslucent',
+  'statusBarBackgroundColor',
+  'darkModeStatusBarBackgroundColor',
+] as const;
+
+type CLIOptionName = typeof AVAILABLE_OPTIONS_NAMES[number];
+
+type CLIOptions = Partial<Record<CLIOptionName, string>>;
+
+interface Configuration {
+  platform: PlatformType;
+  android: AndroidSplashScreenJsonConfig;
+  ios: IosSplashScreenJsonConfig;
+}
+
+async function action(configuration: Configuration) {
+  const { platform, android, ios } = configuration;
+  const rootDir = path.resolve();
+  switch (platform) {
+    case Platform.ANDROID:
+      await configureAndroid(rootDir, android);
+      break;
+    case Platform.IOS:
+      await configureIos(rootDir, ios);
+      break;
+    case Platform.ALL:
+    default:
+      await configureAndroid(rootDir, android);
+      await configureIos(rootDir, ios);
+      break;
+  }
+}
+
+function configurationFromOptions({
+  platform,
+  imageResizeMode,
+  backgroundColor,
+  imagePath,
+  statusBarHidden,
+  statusBarStyle,
+  statusBarTranslucent,
+  statusBarBackgroundColor,
+  darkModeBackgroundColor,
+  darkModeImagePath,
+  darkModeStatusBarStyle,
+  darkModeStatusBarBackgroundColor,
+}: CLIOptions): Configuration {
+  let resolvedPlatform: PlatformType = Platform.ALL;
+  try {
+    resolvedPlatform = (platform && validateEnumValue(platform, Platform)) || Platform.ALL;
+  } catch (e) {
+    throw new Error(`'platform': ${e.message}`);
+  }
+
+  if (!backgroundColor) {
+    throw new Error(
+      `'backgroundColor': Required option is not provided. Provide a valid color string.`
+    );
+  }
+
+  const resolvedStatusBarHidden =
+    statusBarHidden === undefined ? undefined : Boolean(statusBarHidden);
```

Comment:
Passing `undefined` in a variable called `resolved` seems icky to me. Is there a specific reason for passing `undefined` down? Does it say "do not modify anything related to this setting"?

Actually while writing this comment I had deja vu, so if we already talked about this then sorry